src/matlab/homotopy_solver.m

```diff
@@ -86,8 +89,7 @@
     fprintf('\niter\tsigma\t\tcompl_res\tCPU time\tNLP iters\tstatus\n')
 end
 
-
-while (complementarity_iter) > comp_tol && ii < N_homotopy && sigma_k > sigma_N
+while (complementarity_iter) > comp_tol && ii < N_homotopy && (sigma_k > sigma_N || ii == 0)
```

Comment:
why is the ii == 0 part needed?

Reply:
for the direct reformulation  `sigma_k` is set to sigma_N directly. This ensures that even if the user does this that one homotopy iteration will run regardless (which is the intended outcome).